src/checkpoint/checkpoint_txn.c

```diff
@@ -1494,8 +1494,19 @@ __checkpoint_db_internal(WT_SESSION_IMPL *session, const char *cfg[])
      * view of the data, make sure that all the logs are flushed to disk before the checkpoint is
      * complete.
      */
-    if (F_ISSET(&conn->log_mgr, WT_LOG_ENABLED))
+    if (F_ISSET(&conn->log_mgr, WT_LOG_ENABLED)) {
+        /* Crash before metadata sync if checkpoint crash point is configured.
+
+         * FIXME-WT-15069: This is a temporary workaround to allow ckpt_crash_before_metadata_sync
+         * crash point with logging enabled. test/model currently expects crash points to result in
+         * only non-recoverable checkpoints. However, from WT perspective, a crash after the txn
+         * commits is still considered a valid recoverable checkpoint.
```

Comment:
Please make it clear that this crash point should be removed once test/model has been fixed.

Reply:
fixed 22a4cd9f225f2a91e65e4f662a3ca4c16798fa26